### DISCUSSION TITLE
using black

### DISCUSSION THREAD

**Discussion by svinota** [svinota/pyroute2#881] [Ideas]
Proposal:
```
black -C -S -l 79
```
⇒ strings will be left with single comma
⇒ no single member `set`/`tuple`/etc reformatting

Flake8:
```
[flake8]

ignore = E203,E722,W503
```
E203: whitespace around `:` in slices
E722: bare except
W503: line break before binary operator

Rationale: unify the coding standard

Result: >200 files to be reformatted

**Comment by crosser** [comment]
:+1: 

We use `black` now and it made the codebase more readable.

Though the lack of widely accepted line length is making things a bit messy. I for one use `-l79` too.

(And of course the usual advice to keep reformatting commits separate from "real change" commits.)

**Comment by svinota** [comment]
Committed: https://github.com/svinota/pyroute2/commit/a392c67d200d516dedb5990a6a61550922451675

`make format` runs the formatting
`make test-format` checks the formatting

The documentation will follow.
